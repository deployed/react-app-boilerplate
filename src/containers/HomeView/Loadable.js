/**
 * Asynchronously loads the component for HomeView
 */
import Loadable from 'react-loadable';


export default Loadable({
  // we have to provide full path here. It's chunk ID for react loadable
  loader: () => import('containers/HomeView'/* webpackChunkName: "HomeView" */),
  loading: () => null,
});
