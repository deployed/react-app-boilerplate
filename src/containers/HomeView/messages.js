/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage component.
 */
import { defineMessages } from 'react-intl';


export default defineMessages({
  header: {
    id: 'HomePage.header',
    defaultMessage: 'Welcome to Deployed React App Boilerplate',
  },
});
