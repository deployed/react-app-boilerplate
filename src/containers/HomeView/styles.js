import styled, { keyframes } from 'styled-components';

import ReactLogoSVG from './react.svg';


export const Header = styled.div`
  padding: 10px;
  width: 100%;
  height: 200px;
  text-align: center;
  background-color: ${(props) => props.theme.palette.background.default};
  color: white;
  font-family: sans-serif;
`;

const rotate360 = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

export const ReactLogo = styled(ReactLogoSVG)`
  animation: ${rotate360} infinite 20s linear;
  height: 80px;
  width: 80px;
`;
