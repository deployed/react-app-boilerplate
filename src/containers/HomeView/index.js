import React from 'react';
import { FormattedMessage } from 'react-intl';

import messages from './messages';
import { Header, ReactLogo } from './styles';


function Home() {
  return (
    <React.Fragment>
      <Header>
        <ReactLogo />
        <FormattedMessage {...messages.header}>
          {(txt) => (
            <h1>
              {txt}
            </h1>
          )}
        </FormattedMessage>
      </Header>
    </React.Fragment>
  );
}

export default Home;
