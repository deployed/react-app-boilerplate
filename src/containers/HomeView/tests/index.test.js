import React from 'react';
import { shallow } from 'enzyme';

import HomeView from '../index';


describe('<HomeView />', () => {
  it('match snapshot', () => {
    const renderedComponent = shallow(
      <HomeView />,
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
