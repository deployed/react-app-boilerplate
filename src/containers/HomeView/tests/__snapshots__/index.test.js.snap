// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeView /> match snapshot 1`] = `
<React.Fragment>
  <styles__Header>
    <styles__ReactLogo />
    <FormattedMessage
      defaultMessage="Welcome to Deployed React App Boilerplate"
      id="HomePage.header"
      values={Object {}}
    />
  </styles__Header>
</React.Fragment>
`;
